base_layer/core/src/chain_storage/async_db.rs

```diff
@@ -356,6 +356,11 @@ impl<'a, B: BlockchainBackend + 'static> AsyncDbTransaction<'a, B> {
         self
     }
 
+    pub fn update_payref(&mut self, header_hash: HashOutput, output_hash: HashOutput) -> &mut Self {
```

Comment:
```suggestion
    pub fn update_payref(&mut self, header_hash: HashOutputt) -> &mut Self {
```
We can have this be a single tx,should not be multiple,